lib/core/src/sdk.rs

```diff
@@ -1078,12 +1078,54 @@ impl LiquidSdk {
         };
         self.status_stream.track_swap_id(&swap.id)?;
 
-        let accept_zero_conf = swap.get_boltz_create_response()?.accept_zero_conf;
+        let create_response = swap.get_boltz_create_response()?;
+        self.try_lockup(&swap, &create_response).await?;
```

Comment:
It seems that there is still a race where two separate calls to send_payment for the same invoice may end up in broadcasting two lockup transactions. It might happen when both of them are broadcasting in the same time.

Reply:
Yes I agree, I've added a line **TODO** above to fix this. It's similar to Ross' PR

Reply:
I think we can be simpler here than the claim mechanism. Perhaps we only need to mark this payment in memory as in progress and prevent other callers to lockup as long as it is marked as such?

Reply:
Wouldn't adding a `lockup_tx_id` field be the same as marking it?

Reply:
Yes whatever is more simple.